### PR TITLE
feat(messages): allow users to send messages friends only

### PR DESCRIPTION
**Pull request by RiverVanRain**

### CODE REVIEW

**Comment by RiverVanRain** [comment]
Partly solves #3423 #3962

File: mod/messages/views/default/forms/messages/send.php [hunk 1]
```diff
@@ -20,6 +20,7 @@
 		'values' => $recipients,
 		'limit' => 1,
 		'required' => true,
+		'only_friends' => (bool) elgg_get_plugin_setting('friends_only', 'messages') ? true : false,
```
Comment (jeabakker) [inline mod/messages/views/default/forms/messages/send.php:23]: when you cast it to a `bool` there is no need for the `? true : false` part any more ;)

Comment (RiverVanRain) [reply mod/messages/views/default/forms/messages/send.php]: Correct. Fixed

**Comment by jdalsem** [comment]
I want to merge this, but i will try to update this PR for you as it needs rebasing and there are some code style fixes required

**Comment by jdalsem** [comment]
Replaced by https://github.com/Elgg/Elgg/pull/13447

**Pull request closed without merging**